src/Endpoints/Keys.php

```diff
@@ -5,29 +5,133 @@
 namespace MeiliSearch\Endpoints;
 
 use MeiliSearch\Contracts\Endpoint;
+use MeiliSearch\Contracts\Http;
 
 class Keys extends Endpoint
 {
     protected const PATH = '/keys';
 
-    public function get($key): array
+    private ?string $key;
+    private ?string $description;
+    private ?array $actions;
+    private ?array $indexes;
+    private ?string $expiresAt;
+    private ?string $createdAt;
+    private ?string $updatedAt;
+
+    public function __construct(Http $http, $key = null, $description = null, $actions = null, $indexes = null, $expiresAt = null, $createdAt = null, $updatedAt = null)
+    {
+        $this->key = $key;
+        $this->description = $description;
+        $this->actions = $actions;
+        $this->indexes = $indexes;
+        $this->expiresAt = $expiresAt;
+        $this->createdAt = $createdAt;
+        $this->updatedAt = $updatedAt;
+
+        parent::__construct($http);
+    }
+
+    protected function newInstance(array $attributes): self
+    {
+        return new self(
+            $this->http,
+            $attributes['key'],
+            $attributes['description'],
+            $attributes['actions'],
+            $attributes['indexes'],
+            $attributes['expiresAt'],
+            $attributes['createdAt'],
+            $attributes['updatedAt'],
+        );
+    }
+
+    /**
+     * @return $this
+     */
+    protected function fill(array $attributes): self
```

Comment:
I'm not sure how it should be done, but it seems to me like a duplication, what about try to reuse this `fill` with the `newInstance`?

Reply:
Yes it was I remove it